InstaSharper/API/Processors/MediaProcessor.cs

```diff
@@ -115,6 +116,268 @@ internal class MediaProcessor : IMediaProcessor
                 _logger?.LogException(exception);
                 return Result.Fail<bool>(exception);
             }
+        }
+
+        public async Task<IResult<InstaMedia>> UploadVideoAsync(InstaVideo video, InstaImage imageThumbnail, string caption)
+        {
+            try
+            {
+                //POST /api/v1/upload/video/ HTTP/1.1
+                //Host: i.instagram.com
+                //upload_id=1458984764906&media_type=2&_uuid=ba913fe1-5c1a-4d66-b892-b77d16c3906a&_csrftoken=d6662d21cb381d03bd55ddddd6f76aa1
+                var instaUri = UriCreator.GetUploadVideoUri();

```

Comment:
please make sure that your changes at least correctly build, before creating pull request:
```
error CS0117: 'UriCreator' does not contain a definition for 'GetUploadVideoUri'
```